.github/workflows/codacy-analysis.yaml

```diff
@@ -0,0 +1,17 @@
+name: Codacy Analysis CLI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  codacy-analysis-cli:
+    name: Codacy Analysis CLI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@main
+
+      - name: Run Codacy Analysis CLI
+        uses: codacy/codacy-analysis-cli-action@master
```

Comment:
Use a hash code for tagged version instead of branch name; would be good to do for the checkout action too, for convention's sake.